Nullpointer exception in LocalGenerator.java:145

For the following two classes I get a NullPointerException:
```java
import java.io.Serializable;

@SuppressWarnings("serial")
public abstract class Sub<T extends Serializable> {
    protected T value;

    public T getValue() {
        return value;
    }
}
```
and 
```java
import java.util.Comparator;
import java.util.Set;
import java.util.stream.Collectors;

public class SootCrash {

    private <T extends Sub<Long>> String buildFilterExpression(
        final Set<T> ids, String columnName, boolean include) {

        ids.stream()
           .sorted(Comparator.comparingLong(Sub::getValue));

        return null;
    }
}
```

The stack trace is as follows:
```
Caused by: java.lang.NullPointerException
	at soot.javaToJimple.LocalGenerator.generateLocal(LocalGenerator.java:145)
	at soot.LambdaMetaFactory$ThunkMethodSource.unbox(LambdaMetaFactory.java:697)
	at soot.LambdaMetaFactory$ThunkMethodSource.adapt(LambdaMetaFactory.java:646)
	at soot.LambdaMetaFactory$ThunkMethodSource.invokeImplMethod(LambdaMetaFactory.java:798)
	at soot.LambdaMetaFactory$ThunkMethodSource.getInvokeBody(LambdaMetaFactory.java:580)
	at soot.LambdaMetaFactory$ThunkMethodSource.getBody(LambdaMetaFactory.java:419)
	at soot.SootMethod.retrieveActiveBody(SootMethod.java:445)
	at soot.LambdaMetaFactory.makeLambdaHelper(LambdaMetaFactory.java:265)
	at soot.asm.AsmMethodSource.convertInvokeDynamicInsn(AsmMethodSource.java:1477)
	at soot.asm.AsmMethodSource.convert(AsmMethodSource.java:1874)
	at soot.asm.AsmMethodSource.getBody(AsmMethodSource.java:2121)
	... 17 more
```

--- a/src/main/java/soot/LambdaMetaFactory.java
+++ b/src/main/java/soot/LambdaMetaFactory.java
@@ -636,7 +636,7 @@ public final class LambdaMetaFactory {
         // still dealing with
         // booleans semantically.
         // Actually, the wrapper type could be also be other types if wirdcards (?) are used in generic code. 
-        if (wrapper.wrapperTypes.get(from)==null) {
+        if (wrapper.wrapperTypes.get(from) == null) {
           // Insert the cast
           RefType boxedType = wrapper.primitiveTypes.get(to);
           Local castLocal = lc.generateLocal(boxedType);

--- a/src/main/java/soot/LambdaMetaFactory.java
+++ b/src/main/java/soot/LambdaMetaFactory.java
@@ -635,7 +635,8 @@ public final class LambdaMetaFactory {
         // In the example, the map values are of type Object because of generic erasure, but we're
         // still dealing with
         // booleans semantically.
-        if (from == Scene.v().getObjectType()) {
+        // Actually, the wrapper type could be also be other types if wirdcards (?) are used in generic code. 
+        if (wrapper.wrapperTypes.get(from)==null) {
           // Insert the cast
           RefType boxedType = wrapper.primitiveTypes.get(to);
           Local castLocal = lc.generateLocal(boxedType);

